pepper-apis/src/main/java/org/broadinstitute/ddp/route/PatchLastVisitedActivitySectionRoute.java

```diff
@@ -0,0 +1,77 @@
+package org.broadinstitute.ddp.route;
+
+import java.util.Optional;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import org.apache.http.HttpStatus;
+import org.broadinstitute.ddp.constants.ErrorCodes;
+import org.broadinstitute.ddp.constants.RouteConstants;
+import org.broadinstitute.ddp.content.ContentStyle;
+import org.broadinstitute.ddp.db.TransactionWrapper;
+import org.broadinstitute.ddp.db.dao.ActivityInstanceDao;
+import org.broadinstitute.ddp.db.dto.ActivityInstanceDto;
+import org.broadinstitute.ddp.db.dto.LanguageDto;
+import org.broadinstitute.ddp.model.activity.instance.ActivityInstance;
+import org.broadinstitute.ddp.service.ActivityInstanceService;
+import org.broadinstitute.ddp.util.ResponseUtil;
+import org.broadinstitute.ddp.util.RouteUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import spark.Request;
+import spark.Response;
+import spark.Route;
+
+
+public class PatchLastVisitedActivitySectionRoute implements Route {
+
+    private static final Logger LOG = LoggerFactory.getLogger(PatchLastVisitedActivitySectionRoute.class);
+    private ActivityInstanceService actInstService;
+
+    public PatchLastVisitedActivitySectionRoute(ActivityInstanceService actInstService) {
+        this.actInstService = actInstService;
+    }
+
+    @Override
+    public Object handle(Request request, Response response) {
+        String userGuid = request.params(RouteConstants.PathParam.USER_GUID);
+        String studyGuid = request.params(RouteConstants.PathParam.STUDY_GUID);
+        String instanceGuid = request.params(RouteConstants.PathParam.INSTANCE_GUID);
+        JsonElement data = new JsonParser().parse(request.body());
```

Comment:
I recommend subclassing from `ValidatedJsonInputRoute` instead of doing json parsing by hand. Here's the ceremonial boilerplate:

1. Create json class to represent request payload. E.g.
```java
package ...ddp.json;

public class PatchLastVisitedSectionPayload {
  @PositiveOrZero
  private int lastVisitedSection;
  ...
}
```
(Note the annotation `@PositiveOrZero` gives you verification on input for free when you use `ValidatedJsonInputRoute`.)

2. Subclass: `extends ValidatedJsonInputRoute<PatchLastVisitedSectionPayload>`.

3. Override the default:
```java
    @Override
    protected int getValidationErrorStatus() {
        return HttpStatus.SC_BAD_REQUEST;
    }
```

Now in your handler you can just do `payload.getLastVisitedSection()`.